queue_and_stack/dll_queue.py

```diff
@@ -7,13 +7,17 @@ class Queue:
     def __init__(self):
         self.size = 0
         # Why is our DLL a good choice to store our elements?
-        # self.storage = ?
+        self.storage = DoublyLinkedList()
 
     def enqueue(self, value):
-        pass
+        self.storage.add_to_tail(value)
 
     def dequeue(self):
-        pass
+        if not self.storage.length:
+            return None 
+        item = self.storage.remove_from_head()
+        return item
+     
 
     def len(self):
-        pass
+        return self.storage.length
```

Comment:
It's better to use the self.size property of the Queue or STack class to keep track of the length so as to have better control on the data structure.
something like
```
def len(self):
        # pass
        # get number of items in stack or queue
        if self.size > 0:
            return self.size
        else:
            return 0
```